src/components/options/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { h, Component } from 'preact';
+import * as style from './style.scss';
+import { bind } from '../../lib/util';
+import { ImageType, CodecOptions } from '../app';
+
+const CodecConfigurations: {
+  [type: string]: (props: ChildProps) => JSX.Element | null
+} = {
+  original: () => null
+};
+
+type ChildProps = {
+  options: CodecOptions,
+  updateOption(e: Event): void,
+  setOption(key: string, value: any): void
+};
+
+CodecConfigurations.jpeg = ({ options, updateOption }: ChildProps) => (
+    <div>
+        <label>
+            Quality:
+            <input
+                name="quality"
+                type="range"
+                min="1"
+                max="100"
+                step="1"
+                value={options.quality}
+                onInput={updateOption}
+            />
+        </label>
+    </div>
+);
+
+type Props = {
+  name: string,
+  class?: string,
+  type: ImageType,
+  options: CodecOptions,
+  onTypeChange(type: string): void,
+  onOptionsChange(options: any): void
+};
+
+type State = {
+  options: CodecOptions
+};
+
+export default class Options extends Component<Props, State> {
+  state = {
+    options: this.props.options || {}
+  };
+
+  componentWillReceiveProps({ options }: Props) {
+    if (options !== this.props.options) {
+      this.setState({ options });
+    }
+  }
+
+  @bind
+    setType(e: Event) {
```

Comment:
formatting seems wonky here

Reply:
Same goes for other methods in this class

Reply:
Yeah, I just realized I committed this from pixelbook and hadn't installed tslint or eslint in my editor.